src/network/protocol/messages/lib.zig

```diff
@@ -168,3 +168,22 @@ pub fn genericChecksum(m: anytype) [4]u8 {
 
     return digest[0..4].*;
 }
+
+pub fn genericSerialize(m: anytype, allocator: std.mem.Allocator) ![]u8 {
```

Comment:
I thinmyou need comptime checks for `serializeToWriter` and `hintSerializedLen`

Reply:
yes right!